test/ServerRequestFilter/FilterUsingXForwardedHeadersTest.php

```diff
@@ -368,4 +368,30 @@ public function testOnlyHonorsXForwardedProtoIfValueResolvesToHTTPS(
         $uri             = $filteredRequest->getUri();
         $this->assertSame($expectedScheme, $uri->getScheme());
     }
+
+    public function testPortExtractedFromXForwardedHostHeader(): void
+    {
+        $request = new ServerRequest(
+            ['REMOTE_ADDR' => '192.168.1.1'],
+            [],
+            'http://localhost:80/foo/bar',
+            'GET',
+            'php://temp',
+            [
+                'Host'              => 'localhost',
+                'X-Forwarded-Host'  => 'example.com:4433',
```

Comment:
We should probably comment on which kind of server/proxy does this :thinking:

Reply:
It's difficult to me to investigate how exactly `X-FORWARDED-HOST` is made.
In my case `php-fpm` is behind `nginx` which iteself is behind `traefik`. I can't tell which one did that...

Reply:
Are you able to log a full request coming from traefik (in nginx)? Perhaps that will tell you something more.

Practically, we want to give this test a valid name, that explains whether this is Traefik **or** Nginx.

Reply:
I did logging - it comes from **Traefik**.

But I'm not sure it should be used to name a test method though. I guess there are more possible servers that can do the same...
Example at wikipedia shows `host:port` format as possible:
https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Common_non-standard_request_fields

Reply:
Then maybe something along the lines of `testPortCanBeExtractedFromXForwardedHostHeaderWhenPresent()`.